src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/Diagnostics/DiagnosticHandler.cs

```diff
@@ -0,0 +1,310 @@
+
+                var definedSettings = new HashSet<string>();
+
+                foreach (var line in text.Lines)
+                {
+                    var lineText = line.ToString();
+
+                    // Check that setting definition doesn't have more than 1 '='
+                    if (lineText.Where(c => c == '=').Count() > 1)
+                    {
+                        diagnostics.Add(CreateDiagnosticData(document, line, EditorConfigDiagnosticIds.IncorrectSettingDefinition, DiagnosticSeverity.Error, false, 1, Document.Project.Id));
+                    }
+                    else if (lineText.Contains('='))
+                    {
+                        var splitted = lineText.Split('=');
+                        string leftSide = splitted[0].Trim(), rightSide = splitted[1].Replace(" ", "");
+
+                        var settingsSnapshots = SettingsHelper.GetSettingsSnapshots(workspace, filePath);
+                        var settingsItems = settingsSnapshots.Select(sett => sett.GetSettingName()).WhereNotNull();
+
+                        // Check that the setting name exists
+                        if (ShowDiagnosticForSettingName(settingsItems, leftSide))
+                        {
+                            diagnostics.Add(CreateDiagnosticData(document, line, EditorConfigDiagnosticIds.SettingNotFound, DiagnosticSeverity.Error, false, 1, Document.Project.Id));
+                            continue;
+                        }
+
+                        // Check that the setting has not been defined
+                        if (definedSettings.Contains(leftSide))
+                        {
+                            diagnostics.Add(CreateDiagnosticData(document, line, EditorConfigDiagnosticIds.SettingAlreadyDefined, DiagnosticSeverity.Warning, false, 1, Document.Project.Id));
+                        }
+                        definedSettings.Add(leftSide);
+
+                        // Check for settings that define severities
+                        if (rightSide.Contains(':'))
+                        {
+                            var values = rightSide.Split(':');
+                            // Check if it has more than one ':'
+                            if (values.Length > 2)
+                            {
+                                diagnostics.Add(CreateDiagnosticData(document, line, EditorConfigDiagnosticIds.IncorrectSettingDefinition, DiagnosticSeverity.Error, false, 1, Document.Project.Id));
+                                continue;
+                            }
+
+                            // Check if the setting is allowd to define severities
+                            if (!SettingDefinesSeverities(leftSide, settingsSnapshots))
+                            {
+                                diagnostics.Add(CreateDiagnosticData(document, line, EditorConfigDiagnosticIds.SeveritiesNotSupported, DiagnosticSeverity.Error, false, 1, Document.Project.Id));
```

Comment:
Same here, `file_header_template` isn't allowed to have severity, but can contain `:`